### ISSUE TITLE
Referencing AutoLoad scripts fails at random

### ISSUE THREAD

**Issue by leoddd** [godotengine/godot#13841]
_bugsquad edit:_ Additional reproduction project together with reproduction steps is located at: https://github.com/godotengine/godot/issues/13841#issuecomment-348852320

----

**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Windows 10 x64, Godot 3 Beta 1

**Issue description:**

Rarely, maybe 1 out of 20 attempted game launches from the editor, the launch will be interrupted by a popup message claiming that the name of an AutoLoad script is not a valid identifier.
This stops the game from running.

Here is the associated error message in the console:
```
ERROR: GDScriptLanguage::debug_get_stack_level_instance: Condition ' _debug_parse_err_line >= 0 ' is true. returned: 0
   At: modules\gdscript\gdscript_editor.cpp:286
```
Sadly I'm not sure how related this error is, it might just be another symptom of the issue (since Godot can't find the identifier) rather than anything close to the source of the issue. This is just the only new thing that appears in the console every time this happens.

The issue can be temporarily fixed by literally just closing the popup and re-launching the game without even changing anything in the editor, and Godot will no longer complain about the missing identifier for the next while.
This has happened with different AutoLoad scripts for me, so I don't think the issue lies in my script (especially since those aren't changed between Godot accepting them as identifiers and not).

![72ef02f4fafe8d29eccab56871fd6506](https://user-images.githubusercontent.com/20242421/33520387-37f30e00-d7ba-11e7-842f-919b8101fb03.png)
(Note the error message as well as the highlighted message in the script editor.)

**Steps to reproduce:**
Create an AutoLoad script and reference it in code somewhere else using its name. Hit play until you encounter the error.
For example, an AutoLoad `test.gd` with name `boo` set in the Project Settings referenced by using `boo.method()`.

**Comment by Ranoller** [comment]
Probably related with #13444 . The same happend with custom actions. It´s like one time of twenty game don´t want read editor config.

**Comment by leoddd** [comment]
You mention that you only encounter the issue when using F6 to start the project, which might be related to this one since I pretty much exclusively use F6, my project does not even have a main scene defined.
Meaning this could be an issue for F6 launching, yeah.

**Comment by kubecz3k** [comment]
Tested on af27414b1c10173584539186e396668a899e06b2 in Ubuntu 17.04 in release_debug build
I modified Platformer 3D demo to use autoload and I was unable to reproduce this behavior after 40 quick re-runs. 
I would be great if you could you upload some minimal reproduction project to test. The bug might  be platform specific, or dependent somehow on project settings or maybe any other other factors.

**Comment by Ranoller** [comment]
You can test project in #13444 with F6, random fail behavior is similar but with custom actions.

**Comment by leoddd** [comment]
@kubecz3k 
[Autoload Failure.zip](https://github.com/godotengine/godot/files/1525574/Autoload.Failure.zip)

Open this project, open `test.tscn` and hammer F6 until it breaks.
Important note that I figured out after some testing: This does not happen when using F5. Instead, I believe this is related to the bug where the editor will launch the project manager instead of your game, because that occurs with about the same frequency as my autoload issue.

I'm on the official beta 1 and this autoload issue has existed since at least alpha 2, just like how the project manager has been launching instead of your game for at least that long.

That being said, I would emphasize what @Ranoller has been saying; this all seems to occur from Godot refusing to read the project settings on game launch.
At least, I feel like Godot being unable to read the autoload paths would lead to my issue, it being unable to read custom actions leading to @Ranoller's issue, and it being unable to read the default scene path leading to the well-known project-manager-instead-of-game bug.

Might this be related to #13839? Is Godot locking itself out of reading `project.godot`?

**Comment by kubecz3k** [comment]
The error probably is OS specific since i downloaded beta1 on my ubuntu 17.04, I used scene from the comment above and this time I was trying with F6 key. Still no luck in reproducing...

**Comment by Zireael07** [comment]
Most read/save issues seem to be windows specific, just like the one with project manager popping up was.

**Comment by vnen** [comment]
Can you test again with the latest master? This likely have been fixed by #13989.

**Comment by Ranoller** [comment]
I did test this too (with Godot 3 64 bit win compiled at 06/12/2017 at 20:00h) and I´m unable to reproduce the issue (in svevral consecutive executions). This problem was definitely fixed.
